### PR TITLE
(fix) Backend: Modified delete functionality to delete from mongo and elastic.

### PR DESCRIPTION
**Pull request by mdhaman**

### CODE REVIEW

**Comment by sivakuna-aap** [comment]
:+1:

File: server/superdesk/io/commands/remove_expired_content.py [hunk 1]
```diff
@@ -80,14 +81,14 @@ def get_expired_items(provider_id, expiration_date):
...
 
-def remove_items_from_elastic(ids):
-    batch_size = 500
-    print("total documents to be removed {}".format(len(ids)))
-    for i in range(0, len(ids), batch_size):
-        batch = ids[i:i + batch_size]
-        query = {'query': {'terms': {'ingest._id': batch}}}
-        superdesk.app.data._search_backend('ingest').remove('ingest', query)
-        print("Removed {} documents from ingest.".format(len(batch)))
+# def remove_items_from_elastic(ids):
```
Comment (amagdas) [inline server/superdesk/io/commands/remove_expired_content.py]: Remove please...

File: server/superdesk/eve_backend.py [hunk 2]
```diff
@@ -120,13 +123,37 @@ def replace_in_search(self, endpoint_name, id, document, original):
...
+        :param endpoint_name: name of the endpoint
+        :param ids: list of ids
+        """
+        ids = [str(doc_id) for doc_id in ids]
+        batch_size = 500
+        logger.info("total documents to be removed {}".format(len(ids)))
+        for i in range(0, len(ids), batch_size):
+            batch = ids[i:i + batch_size]
+            query = {'query': {'terms': {'{}._id'.format(endpoint_name): batch}}}
+            app.data._search_backend(endpoint_name).remove(endpoint_name, query)
```
Comment (amagdas) [inline server/superdesk/eve_backend.py]: Is eve-elastic returning any result after deletion? We should be able to compare the number of deleted items and log a warning/error if they don't match...

File: server/superdesk/io/commands/remove_expired_content.py [hunk 3]
```diff
@@ -63,8 +64,8 @@ def remove_expired_data(provider):
 
     if ids:
         print('Removing items %s' % ids)
-        superdesk.app.data._backend('ingest').remove('ingest', {'_id': {'$in': ids}})
-        remove_items_from_elastic(ids)
+        ingest_service.delete({'_id': {'$in': ids}})
+        # remove_items_from_elastic(ids)
```
Comment (amagdas) [inline server/superdesk/io/commands/remove_expired_content.py]: Please remove...

File: server/superdesk/eve_backend.py (hunk 2 above)
Comment (mdhaman) [inline server/superdesk/eve_backend.py]: Elasticsearch will return following status {'_indices': {'sptest': {'_shards': {'failed': 0, 'total': 5, 'successful': 5}}}}.
In Lucene, it just marks the record as deleted and there is no api to give the deleted record count.

**Comment by amagdas** [comment]
:+1:

**Comment by mugurrus** [comment]
:+1:

**Pull request merged**